feature-libs/pickup-in-store/components/container/cart-pickup-options-container/cart-pickup-options-container.component.ts

```diff
@@ -4,44 +4,157 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { Component, OnInit, Optional } from '@angular/core';
-import { OrderEntry } from '@spartacus/cart/base/root';
 import {
+  Component,
+  ElementRef,
+  OnInit,
+  Optional,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import { ActiveCartFacade, OrderEntry } from '@spartacus/cart/base/root';
+import { PreferredStoreService } from '@spartacus/pickup-in-store/core';
+import {
+  IntendedPickupLocationFacade,
   PickupLocationsSearchFacade,
   PickupOption,
+  SetPickupOptionDeliveryPayload,
 } from '@spartacus/pickup-in-store/root';
-import { OutletContextData } from '@spartacus/storefront';
-import { Observable } from 'rxjs';
-import { filter, map, switchMap, tap } from 'rxjs/operators';
+import {
+  LaunchDialogService,
+  OutletContextData,
+  LAUNCH_CALLER,
+} from '@spartacus/storefront';
+import { Observable, Subscription } from 'rxjs';
+import { filter, map, switchMap, take, tap } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-cart-pickup-options-container',
   templateUrl: 'cart-pickup-options-container.component.html',
 })
 export class CartPickupOptionsContainerComponent implements OnInit {
+  @ViewChild('open') element: ElementRef;
+  subscription = new Subscription();
   pickupOption$: Observable<PickupOption>;
   displayName$: Observable<string>;
+  cartId: string;
+  name: string;
+  entryNumber: number;
+  productCode: string;
+  quantity: number;
+  userId: string;
+  displayPickupLocation$: Observable<string | undefined>;
+  availableForPickup = false;
+  private displayNameIsSet = false;
+
   constructor(
     @Optional() protected outlet: OutletContextData<OrderEntry>,
-    protected storeDetails: PickupLocationsSearchFacade
+    protected activeCartFacade: ActiveCartFacade,
+    protected intendedPickupLocationService: IntendedPickupLocationFacade,
+    protected launchDialogService: LaunchDialogService,
+    protected preferredStoreService: PreferredStoreService,
+    protected pickupLocationsSearchService: PickupLocationsSearchFacade,
+    protected vcr: ViewContainerRef
   ) {
     // Intentional empty constructor
   }
 
   ngOnInit() {
     this.pickupOption$ = this.outlet?.context$.pipe(
+      filter((outletContextData) => !!outletContextData),
+      tap(
+        (outletContextData) =>
+          (this.entryNumber = outletContextData.entryNumber ?? -1)
+      ),
+      tap(
+        (outletContextData) =>
+          (this.quantity = outletContextData.quantity ?? -1)
+      ),
+      tap(
+        (outletContextData) =>
+          (this.productCode = outletContextData.product?.code || '')
+      ),
+      tap(
+        (outletContextData) =>
+          (this.name = outletContextData.deliveryPointOfService?.name ?? '')
+      ),
```

Comment:
Do all of these need to be in separate taps? It'd be shorter and perhaps more readable in one.

```suggestion
      tap(
        (outletContextData) => {
          this.entryNumber = outletContextData.entryNumber ?? -1;
          this.quantity = outletContextData.quantity ?? -1;
          this.productCode = outletContextData.product?.code || '';
          this.name = outletContextData.deliveryPointOfService?.name ?? '';
      ),
```